pkg/staticnetworkconfig/generator.go

```diff
@@ -177,7 +180,13 @@ func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistence(macInterf
 		connectionSection := cfg.Section("connection")
 		interfaceName, err := connectionSection.GetKey("interface-name")
 		if err != nil {
-			return errors.Wrapf(err, "failed to get interface-name for file %s", nc.FilePath)
+			interfaceID, err := connectionSection.GetKey("id")
+			if err == nil {
+				s.log.Infof("interface %s found using identifier, mapping not required", interfaceID.String())
+				matched = true
```

Comment:
1. Please move the change to a separate commit with informative commit message explaining why it is done (support mac-identifier). BTW, I think the change in this case should be just ignoring the error and `continue`.
2. Regarding the `matched = true` added here-  `matched` is used to mark that at least one interface in the yaml/connection files matches an entry in the MAC map. Here there was no such matching.
3. Regarding `match` logic in general - As @zaneb mentioned, the "matched" logic is breaking ABI (which in some cases has only a [single](https://github.com/openshift/installer/blob/master/pkg/asset/agent/agentconfig/agenthosts.go#L260-L267) MAC address in the MAC map, that has no corresponding interface in the yaml). We need to remove the `matched` logic from the method here and add a separate validation, verifying the MAC map is not empty. Same here, separate commit please.

Reply:
Moved the check for a entry in interface table into a separate function. Split changes into separate commits.